src/App/Fossa/Analyze.hs

```diff
@@ -158,14 +158,15 @@ tryUploadContributors baseDir baseUrl apiKey locator = do
   contributors <- fetchGitContributors baseDir
   uploadContributors baseUrl apiKey locator contributors
 
-fossaProjectUrl :: URI -> Text -> Text -> Text
-fossaProjectUrl baseUrl rawLocator branch = URI.render baseUrl <> "projects/" <> encodedProject <> "/refs/branch/" <> branch <> "/" <> encodedRevision
+fossaProjectUrl :: URI -> Text -> ProjectRevision -> Text
+fossaProjectUrl baseUrl rawLocator revision = URI.render baseUrl <> "projects/" <> encodedProject <> "/refs/branch/" <> branch <> "/" <> encodedRevision
   where
     Locator{locatorFetcher, locatorProject, locatorRevision} = parseLocator rawLocator
 
     underBS :: (ByteString -> ByteString) -> Text -> Text
     underBS f = TE.decodeUtf8 . f . TE.encodeUtf8
 
+    branch = underBS (urlEncode True) (fromMaybe undefined (projectBranch revision))
```

Comment:
How do we resolve this undefined?  The URL here requires a branch.

Reply:
Good question. I wonder if there's a URL we can use that doesn't require a branch. If there isn't, we'll have to hardcode what core uses (`master`) as the default 😞

Reply:
In which case, we should just resolve the `Maybe` branch name earlier in the chain, to take advantage of that default everywhere.